test/functional/provider/ruby_spec.lua

```diff
@@ -0,0 +1,69 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local eq = helpers.eq
+local neq = helpers.neq
+local feed = helpers.feed
+local clear = helpers.clear
+local funcs = helpers.funcs
+local meths = helpers.meths
+local insert = helpers.insert
+local expect = helpers.expect
+local command = helpers.command
+local exc_exec = helpers.exc_exec
+local write_file = helpers.write_file
+local curbufmeths = helpers.curbufmeths
```

Comment:
There should be a check like https://github.com/justinmk/neovim/blob/master/test/functional/provider/python_spec.lua#L16-L26 so that the tests show up in the "pending" list (and won't fail) if ruby is missing.

```lua
  do
    clear()
    command('let [g:interp, g:errors] = provider#ruby#Detect(2)')
    local errors = meths.get_var('errors')
    if errors ~= '' then
      pending(
        'Ruby (or the Ruby neovim gem) is broken or missing:\n' .. errors,
        function() end)
      return
    end
  end
```